lib/repl.js

```diff
@@ -283,17 +283,23 @@ function REPLServer(prompt,
   self._domain.on('error', function debugDomainError(e) {
     debug('domain error');
     const top = replMap.get(self);
+
     internalUtil.decorateErrorStack(e);
+    const isError = internalUtil.isError(e);
     if (e instanceof SyntaxError && e.stack) {
       // remove repl:line-number and stack trace
       e.stack = e.stack
                  .replace(/^repl:\d+\r?\n/, '')
                  .replace(/^\s+at\s.*\n?/gm, '');
-    } else if (e.stack && self.replMode === exports.REPL_MODE_STRICT) {
+    } else if (isError && self.replMode === exports.REPL_MODE_STRICT) {
       e.stack = e.stack.replace(/(\s+at\s+repl:)(\d+)/,
                                 (_, pre, line) => pre + (line - 1));
     }
-    top.outputStream.write((e.stack || e) + '\n');
+    if (isError && e.stack) {
+      top.outputStream.write(`${e.stack}\n`);
+    } else {
+      top.outputStream.write(`Error: ${e}\n`);
```

Comment:
@TimothyGu yes, but I think `String` is a good middle ground - I don't want it to be bulletproof - I want it to be useful for users.